eventuals/generator.h

```diff
@@ -16,15 +18,32 @@ namespace eventuals {
 
 ////////////////////////////////////////////////////////////////////////
 
-template <typename E_, typename From_, typename To_>
+template <
+    typename E_,
+    typename From_,
+    typename To_,
+    typename Catches_,
+    typename Raises_>
 struct HeapGenerator final {
+  using BeginCallback_ = Callback<void(TypeErasedStream&)>;
+  using FailCallback_ = Callback<function_type_t<
+      void,
+      get_rvalue_type_or_void_t<
+          typename VariantErrorsHelper<
+              variant_of_type_and_tuple_t<
+                  std::monostate,
+                  Raises_>>::type>>>;
+  using StopCallback_ = Callback<void()>;
+  using BodyCallback_ = Callback<function_type_t<void, To_>>;
+  using EndedCallback_ = Callback<void()>;
```

Comment:
Let's move these to the top level and call them `GeneratorBeginCallback` (without the trailing underscore) and make them templated if necessary (e.g.,  `GeneratorFailCallback` will take a `Raises` template parameter) and then we can use them throughout this file without having to create them 3 places!